app/views/products/statistics.html.slim

```diff
@@ -0,0 +1,33 @@
+= stylesheet_link_tag 'statistics'
+
+- total = 0
+- categories_total = Hash.new
+- @categories.each do |category|
+  - categories_total[category.name] = 0
+- @products.each do |product|
+  - total += product.price * product.quantity
+  - categories_total[product.category] += product.price * product.quantity
```

Comment:
Można stworzyć 2 metody klasowe (statyczne) w modelu:
```ruby
def self.total_price
  total = 0
  all.each { |product| total += product.price * product.quantity }
  total
end

def self.total_price_by_category
  total = {}
  all.each do |product|
    total[product.category] ||= []
    total[product.category] += product.price * product.quantity
  end
  total
end
```

i wtedy użyć 
```ruby
= 'Total: ' + ("%.2f" % [@products.total_price])
```
oraz
```ruby
- @products.total_price_by_category.each do |key, value|
  div
    = key + ": "+("%.2f" % [value])
```

Reply:
Generalnie chodzi o to, że kiedy ta cała logika jest w widoku, trudno jest to przetestować testami automatycznymi. Kiedy przeniesiemy metody do modelu, możemy je łatwo przetestować i używać w wielu miejscach